ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -826,29 +828,47 @@ private class PostgresLedgerDao(
           Option[WorkflowId],
           Date,
           InputStream,
-          Option[InputStream]))(implicit conn: Connection) =
+          Option[InputStream],
+          InputStream))(implicit conn: Connection) =
     contractResult match {
-      case (coid, transactionId, workflowId, ledgerEffectiveTime, contractStream, keyStreamO) =>
+      case (coid, transactionId, workflowId, ledgerEffectiveTime, contractStream, keyStreamO, tx) =>
         val witnesses = lookupWitnesses(coid)
         val divulgences = lookupDivulgences(coid)
+        val absoluteCoid = AbsoluteContractId(coid)
+
+        val (signatories, observers) =
+          transactionSerializer
+            .deserializeTransaction(ByteStreams.toByteArray(tx))
+            .getOrElse(sys.error(s"failed to deserialize transaction! cid:$coid"))
+            .nodes
+            .collectFirst {
+              case (_, NodeCreate(coid, _, _, signatories, stakeholders, _))
+                  if coid == absoluteCoid =>
+                (signatories, stakeholders diff signatories)
+            } getOrElse {
+            logger.warn(s"Unable to read stakeholders for contract $coid, returning empty result")
```

Comment:
Should this really be a warning? About what is the reader of the log being warned? What can they do, once they read the warning? Do we always expect this data to be available?

Reply:
If I'm not mistaken the most likely occurrence of this would be asking data to a ledger version that did not expose stakeholders. If that's the case, we can probably lower the log level, maybe to debug. Thoughts?

Reply:
Should we consider this to be an actual error (i.e. crash)?

Reply:
As discussed offline, I was thinking of the Ledger API, where these fields could have not been available. Since we're reading the transaction directly, this is clearly an error condition.

Reply:
Fixed by https://github.com/digital-asset/daml/pull/2316/commits/47b714a3ed00ebad6f3ae49f66e85e66dbcf2c20